RecursiveExtractor.Tests/ExtractorTests/DisposeBehaviorTests.cs

```diff
@@ -121,4 +122,51 @@ await foreach (var file in results)
             Assert.ThrowsException<ObjectDisposedException>(() => disposedResult.Content.Position);
         }
     }
+
+    [DataTestMethod]
+    [DataRow("TestData.zip")]
+    public void EnsureDisposedWithExtractToDirectory(string fileName)
+    {
+        var directory = TestPathHelpers.GetFreshTestDirectory();
+        var copyDirectory = TestPathHelpers.GetFreshTestDirectory();
+        Directory.CreateDirectory(copyDirectory);
+        var path = Path.Combine(Directory.GetCurrentDirectory(), "TestData", "TestDataArchives", fileName);
+        // Make a copy of the archive so it can be deleted later to confirm properly disposed
+        var copyPath = Path.Combine(copyDirectory, fileName);
+        File.Copy(path, copyPath);
+        var extractor = new Extractor();
+        extractor.ExtractToDirectory(directory, copyPath);
+        File.Delete(copyPath);
+        if (Directory.Exists(directory))
+        {
+            Directory.Delete(directory, true);
+        }
+        if (Directory.Exists(copyDirectory)) {
+            Directory.Delete(copyDirectory, true);
+        }
```

Comment:
It isn't clear to me if there's an assertion in this test or in `EnsureDisposedWithExtractToDirectoryAsync`.

Reply:
There isn't an assertion here - the test fails if the Stream handle isn't properly disposed which would cause the Delete call to throw an exception. But that does raise the issue that if the test *does* fail the copy (and extracted files) won't be cleaned up so probably worth adding a bit more to the test - maybe a cleanup at the class level.